zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- The existing consensus rule requiring transactions that spend coinbase outputs to have
+  an empty ``vout``, is amended to only apply to transactions that spend transparent
+  coinbase outputs.
+
+- The existing consensus rule requiring coinbase outputs to have 100 confirmations before
+  they may be spent (coinbase maturity), is amended to only apply to transparent coinbase
+  outputs.
+
+- All Sapling outputs in coinbase transactions MUST have valid note commitments when
+  recovered using a 32-byte array of zeroes as the outgoing viewing key.
+
+
+Interaction with funding streams
+--------------------------------
+
+If ZIP 207 [#zip-0207]_ is accepted and deployed prior to (or in the same network upgrade
+as) this ZIP, the following additional consensus rules are enforced once the TODO network
+upgrade activates:
+
+- Funding streams MAY target Sapling addresses in addition to transparent P2SH addresses.
+
+- The existing consensus rule from ZIP 207 [#zip-0207]_ requiring coinbase transactions
+  for block height ``height`` to pay active funding streams with a standard P2SH script,
+  is amended to only apply to active funding streams that target a transparent address at
+  ``height``.
+
+- In the coinbase transaction for block height ``height``, for every active funding stream
+  that targets a Sapling address at ``height``, the transaction MUST include at least one
+  Sapling output that pays exactly the funding stream's value to the target Sapling
+  address.
+
+
+Interaction with the Founders' Reward
+-------------------------------------
+
+This ZIP does not alter the existing Founders' Reward addresses. If a future ZIP proposed
+moving to shielded Founders' Reward addresses (instead of, or before, deploying ZIP 207),
+the necessary consensus rules would be similar to those specified above for funding
+streams.
+
+
+Rationale
+=========
+
+The ZIP does not require that all coinbase must be shielded immediately from activation of
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because the Founders' Reward targets transparent addresses, and thus it remains
+necessary for the time being to support them. A future ZIP could require all coinbase to
+be shielded immediately.
+
+Enforcing coinbase maturity at the consensus level for Sapling outputs would incur
+significant complexity in the consensus rules, because it would require special-casing
+coinbase note commitments in the Sapling commitment tree. The standard recommendation when
+spending a note is to select an anchor 10 blocks back from the current chain tip, which
+acts as a de-facto 10-block maturity on all notes, coinbase included. This might be
+proposed as a consensus rule in future.
```

Comment:
Wouldn't this break a lot of functionality as a consensus rule?

Reply:
Not that I can see. It simply forces users to wait an extra ten blocks before spending newly-received ZEC, which from a usability perspective is equivalent to the transaction being mined 10 blocks "late" (something apps already have to account for in their UX, because this can happen naturally on the network today).